arduino/examples/ESP32IOBoard/ESP32IOBoard.ino

```diff
@@ -0,0 +1,134 @@
+/** \copyright
+ * Copyright (c) 2019, Mike Dunston
+ * All rights reserved.
+ *
+ * Redistribution and use in source and binary forms, with or without
+ * modification, are  permitted provided that the following conditions are met:
+ * 
+ *  - Redistributions of source code must retain the above copyright notice,
+ *    this list of conditions and the following disclaimer.
+ *
+ *  - Redistributions in binary form must reproduce the above copyright notice,
+ *    this list of conditions and the following disclaimer in the documentation
+ *    and/or other materials provided with the distribution.
+ *
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS"
+ * AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+ * IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE
+ * ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT HOLDER OR CONTRIBUTORS BE
+ * LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR
+ * CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF
+ * SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS
+ * INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN
+ * CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE)
+ * ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+ * POSSIBILITY OF SUCH DAMAGE.
+ *
+ * \file ESP32IOBoard.ino
+ * 
+ * Main file for the io board application on an ESP32.
+ * 
+ * @author Mike Dunston
+ * @date 13 January 2019
+ */
+
+#include <Arduino.h>
+#include <WiFi.h>
+#include <ESPmDNS.h>
+#include <vector>
+
+#include <OpenMRN.h>
+#include <openlcb/TcpDefs.hxx>
+#include "config.hxx"
+
+constexpr uint16_t OPENMRN_TCP_PORT = 12021L;
+
+WiFiServer openMRNServer(OPENMRN_TCP_PORT);
+
+const char* ssid     = "apname";
+const char* password = "password";
+const char* hostname = "esp32mrn";
+
+static constexpr uint64_t NODE_ID = UINT64_C(0x050101011423);
+OpenMRN openmrn(NODE_ID);
+
+// ConfigDef comes from config.hxx and is specific to the particular device and
+// target. It defines the layout of the configuration memory space and is also
+// used to generate the cdi.xml file. Here we instantiate the configuration
+// layout. The argument of offset zero is ignored and will be removed later.
+openlcb::ConfigDef cfg(0);
+
+#if !defined(LED_BUILTIN)
+constexpr uint8_t LED_BUILTIN = 13;
+#endif
+
+GPIO_PIN(LED, GpioOutputSafeLow, LED_BUILTIN);
+
+openlcb::ConfiguredConsumer onboard_led(
+    openmrn.stack()->node(), cfg.seg().consumers().entry<0>(), LED_Pin());
+
+class WiFiClientAdapter {
```

Comment:
My plan is once I get this working to move it under freertos_drivers/arduino/WiFiGridConnectAdapter.hxx or something along those lines. I'd also like to explore the direct usage of LwIP APIs instead of using WiFiServer/WiFiClient. But yes this will get refactored out of the example and into a more generic approach.